core/io/resource_loader.cpp

```diff
@@ -242,16 +242,18 @@ ResourceLoader::LoadToken::~LoadToken() {
 }
 
 Ref<Resource> ResourceLoader::_load(const String &p_path, const String &p_original_path, const String &p_type_hint, ResourceFormatLoader::CacheMode p_cache_mode, Error *r_error, bool p_use_sub_threads, float *r_progress) {
+	String original_path = p_original_path.is_empty() ? p_path : p_original_path;
```

Comment:
```suggestion
	const String &original_path = p_original_path.is_empty() ? p_path : p_original_path;
```

Reply:
I just realized I may be a good idea to reason about this change request. As you may already know, `String` is backed by a copy-on-write container, with a reference counter. This counter is updated atomically, which may or may not be bad for performance depending on how hot the code path is, CPU architecture, etc. In any case, my motto here is, "Why taking the risk?". If you know you just want to keep the temporary string alive, a const reference would do that without involving a copy, which would deal with the refcount.